From ebceefcc56614d3a079c5b905bcba5a1797b60c4 Mon Sep 17 00:00:00 2001
From: Julian Sikorski <belegdol+github@gmail.com>
Date: Sat, 27 Jan 2024 13:27:31 +0000
Subject: [PATCH] Revert "ODROID-N2: net: stmmac: enable/disable WOL irq"

This reverts commit fa6baada11bc2aaa35ce26b97380f37aab14c1dd.
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_ethtool.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_ethtool.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_ethtool.c
index 77fb5598b..a4b5636c9 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_ethtool.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_ethtool.c
@@ -761,12 +761,10 @@ static int stmmac_set_wol(struct net_device *dev, struct ethtool_wolinfo *wol)
 	if (wol->wolopts) {
 		pr_info("stmmac: wakeup enable\n");
 		device_set_wakeup_enable(priv->device, 1);
-		if (priv->wol_irq != dev->irq)
-			enable_irq_wake(priv->wol_irq);
+		enable_irq_wake(priv->wol_irq);
 	} else {
 		device_set_wakeup_enable(priv->device, 0);
-		if (priv->wol_irq != dev->irq)
-			disable_irq_wake(priv->wol_irq);
+		disable_irq_wake(priv->wol_irq);
 	}
 
 	mutex_lock(&priv->lock);
-- 
2.34.1

